### PR TITLE
#170947575 booking an accommodation

### PR DESCRIPTION
**Pull request by Baraka-Mugisha**
#### What does this PR do?
add a `user can book an accommodation facility` feature

#### Description of Task to be completed?
- added tests for the feature
- added a controller
- validated all inputs
- documented the feature
- internationalized all messages
- added booking model and migration

#### How should this be manually tested?
- clone this repo
- run npm install to install all dependencies
-  run the server (npm run reset:dev)
- make a put request to `/api/v1/facilities/book`

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#170947575](https://www.pivotaltracker.com/story/show/170947575)

#### Screenshots (if appropriate)
Screen Shot 2020-02-27 at 16 14 49

#### Questions:
- N/A

### CODE REVIEW

**Review by bbaime98 — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]
LGTM

File: src/index.js [hunk 1]
```diff
@@ -46,6 +46,11 @@ app.use('/api/v1/users', userRouter);
 app.use('/api/v1/trips', tripsRouter);
 app.use('/api/v1/facilities', facilitiesRouter);
 
+
+app.use((req, res) => res.status(404).send({ status: 404, error: `Route ${req.url} Not found.` }));
```
Comment (izzett222) [inline src/index.js]: hey @Baraka-Mugisha , could you add the internalization to this response

**Review by izzett222 — DISMISSED** [review]

File: src/index.js (hunk 1 above)
Comment (Baraka-Mugisha) [reply src/index.js]: It is done

**Review by izzett222 — DISMISSED** [review]

**Review by bbaime98 — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]

File: src/controllers/tripsController.js [hunk 2]
```diff
@@ -66,31 +61,26 @@ export default class requestController {
...
+      const { user } = req;
+      const {
+        location,
+        destination,
+        departureDate,
+        returnDate,
+        reason,
+        gender, role, passportName
+      } = req.body;
+      if (Date.parse(departureDate) > Date.parse(returnDate)) {
```
Comment (ceelogre) [inline src/controllers/tripsController.js]: If the dates are equal, it means. I can book an accommodation where I won't spend a night. Is this the intended result?

File: src/controllers/tripsController.js [hunk 3]
```diff
@@ -32,29 +31,25 @@ export default class requestController {
...
-            role
-          }],
-          status: 'open',
-        });
-        return Response.success(res, 201, 'Request created successfully', newRequest);
-      }
+
+      const newRequest = await db.Request.create({
+        id: uuid(),
+        type: 'one way',
```
Comment (ceelogre) [inline src/controllers/tripsController.js:37]: 'One way' is a `type`, it should be an enum or some other non-editable form.

File: src/controllers/tripsController.js [hunk 4]
```diff
@@ -32,29 +31,25 @@ export default class requestController {
...
+        destination,
+        reason,
+        departureDate,
+        email,
+        profileData: [{
+          gender,
+          passportName,
+          role
+        }],
+        status: 'open',
```
Comment (ceelogre) [inline src/controllers/tripsController.js]: 'Open' should also be some non-editable form.

File: src/index.js [hunk 5]
```diff
@@ -39,13 +39,17 @@ app.use(passport.session());
...
+app.use('/', welcome);
 app.use('/api-doc', swagger);
 app.use('/api/v1/auth', authRouter);
 app.use('/api/v1/users', userRouter);
 app.use('/api/v1/trips', tripsRouter);
 app.use('/api/v1/facilities', facilitiesRouter);
 
+app.use((req, res) => res.status(404).send({ status: 404, error: res.__('Route %s not found', req.url) }));
+
+app.use((err, req, res) => res.status(500).send({ status: 500, error: err }));
```
Comment (ceelogre) [inline src/index.js]: Why are you sending technical error messages to the user?

**Review by ceelogre — CHANGES_REQUESTED** [review]
Take a look at the comments, also I've not seen unit tests.

File: src/index.js (hunk 5 above)
Comment (Baraka-Mugisha) [reply src/index.js]: We removed the message

File: src/controllers/tripsController.js (hunk 2 above)
Comment (Baraka-Mugisha) [reply src/controllers/tripsController.js]: We changed it to allowing a user to spend at least one night!!

File: src/index.js [hunk 6]
```diff
@@ -39,13 +39,17 @@ app.use(passport.session());
...
+app.use('/', welcome);
 app.use('/api-doc', swagger);
 app.use('/api/v1/auth', authRouter);
 app.use('/api/v1/users', userRouter);
 app.use('/api/v1/trips', tripsRouter);
 app.use('/api/v1/facilities', facilitiesRouter);
 
+app.use((req, res) => res.status(404).send({ status: 404, error: res.__('Route %s not found', req.url) }));
+
+app.use((err, req, res) => res.status(500).send({ status: 500, error: res.__('SERVER ERROR') }));
```
Comment (izzett222) [inline src/index.js]: can you change the response message to lowercase
```suggestion
app.use((err, req, res) => res.status(500).send({ status: 500, error: res.__('server error') }));
```

**Review by izzett222 — CHANGES_REQUESTED** [review]

File: src/migrations/20200226141733-create-bookings.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.dropTable('Bookings');
+  }
+};
```
Comment (izzett222) [inline src/migrations/20200226141733-create-bookings.js]: could you add an empty line for this file. @Baraka-Mugisha

**Review by izzett222 — DISMISSED** [review]

**Review by bbaime98 — DISMISSED** [review]

File: src/migrations/20200226141733-create-bookings.js (hunk 7 above)
Comment (Baraka-Mugisha) [reply src/migrations/20200226141733-create-bookings.js]: I added it

File: src/index.js (hunk 6 above)
Comment (Baraka-Mugisha) [reply src/index.js]: I changed it.

**Review by izzett222 — APPROVED** [review]

**Review by bbaime98 — APPROVED** [review]

**Review by ceelogre — APPROVED** [review]

**Pull request merged**